storage/src/backend/registry.rs

```diff
@@ -394,18 +462,35 @@ impl RegistryReader {
         offset: u64,
         allow_retry: bool,
     ) -> RegistryResult<usize> {
+        let state = self.get_state();
         let url = format!("/blobs/sha256:{}", self.blob_id);
-        let url = self
-            .state
-            .url(url.as_str(), &[])
-            .map_err(RegistryError::Url)?;
+        let url = state.url(url.as_str(), &[]).map_err(RegistryError::Url)?;
         let mut headers = HeaderMap::new();
         let end_at = offset + buf.len() as u64 - 1;
         let range = format!("bytes={}-{}", offset, end_at);
         headers.insert("Range", range.parse().unwrap());
 
+        if !self.is_registry() {
+            let mirror_index = self.mirror_state.index.load().unwrap();
+            let mirror = &self.mirror_state.mirrors[mirror_index];
+
+            for (key, value) in &mirror.config.headers {
+                let value =
+                    if key == "X-Dragonfly-Registry" && !value.starts_with(&self.state.scheme) {
+                        format!("{}://{}", self.state.scheme, value)
```

Comment:
Why replace the target registry scheme, I think user can specify a different one with `state.scheme`.

Reply:
This code is to resolve the case of `X-Dragonfly-Registry` without scheme in the configuration file. Do you mean the scheme has to be filled in here? (as containerd does)

Reply:
Add `state.scheme` as scheme only when there is no scheme?